pom.xml

```diff
@@ -364,9 +364,9 @@
         </configuration>
         <executions>
           <execution>
-            <id>format</id>
+            <id>format-sources</id>
             <goals>
-              <goal>check</goal>
+              <goal>apply</goal>
```

Comment:
I have mixed feelings here, I see pros and cons in going down this path.
The current setup already prevents committing code that is not formatted, as it will break in CI.

I defer the final decision to someone else in the team.

Reply:
Let's wait for 3rd opinion and I will split PR to smaller as needed.

Reply:
> recently working with Golang team I found it is so natural to auto-format code

I have to appreciate that there is a bit of a gap in-between Go formatting IDE/Editors integration and Java, but I'm fine either way.

Reply:
I don't have a strong opinion either. I'm comfortable with the Go style of doing things. But Go also has a single, simple formatter that ships with it.

I lean towards saying this is positive, though small. I think it's more likely for people to be irritated that their PR fails because of code formatting than they would be irritated that the code gets auto-formatted under their feet.